[WIP] dom0: add possibility to use Domu without GSX

rshym opened the pull request:
The 'gsx' MACHINE_FEATURE is used to generate proper configuration files for domains.

rshym committed:
yaml: add `console` option for `--GRAPHICS` parameter

This commit adds console only build for DomD and DomU.

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>

rshym commented:
Successfully built and ran on `h3ulcb-4x2g` and `h3ulcb-4x2g-kf` with `console` and `sources`.

lorc commented on meta-xt-prod-devel-rcar-control/recipes-guests/domu/domu.bbappend:
```diff
@@ -44,8 +45,12 @@ do_install:append() {
-        cat ${WORKDIR}/domu-vdevices.cfg >> ${CFG_FILE}
-        cat ${WORKDIR}/pvr-${XT_DOMU_CONFIG_NAME} >> ${CFG_FILE}
+        if ${@bb.utils.contains('MACHINE_FEATURES', 'gsx', 'true', 'false', d)}; then
+            cat ${WORKDIR}/domu-vdevices.cfg >> ${CFG_FILE}
```
I'd better removed `gsx` from `domu-vdevices` and added `domu-vdevices-gsx. Idea is that you always include `domu-vdevices` and add `domu-devices-gsx` conditionally.

lorc commented on meta-xt-prod-devel-rcar-control/recipes-guests/domu/files/domu-vdevices-nogsx.cfg:
```diff
@@ -0,0 +1,24 @@
+# Note2:
+# Option `target` must be last element in list. For explanation see
+# https://xenbits.xenproject.org/docs/4.15-testing/man/xl-disk-configuration.5.html
+disk = [ 'backend=DomD, vdev=xvda1, target=/dev/STORAGE_PART3' ]
```
... approach described in the other command will ensure that there will be duplicate entries like this.

rshym committed:
dom0: add possibility to use DomU without GSX

GSX-related configs are separated and are moved
into gsx-common.cfg.
The SOC-specific GSX options are added at compile time.

The 'gsx' MACHINE_FEATURE is required to generate proper
configuration files for domains.

Signed-off-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>

rshym commented:
I have reworked PR.
- domu-vdevices.cfg is reworked
- GSX-related part of domu.cfg is moved into gsx-common.cfg
- the python function selects SOC-specific GSX-related part of the domu.cfg
- m3n related config is removed (I have checked available products, and didn't find usage of that SOC)

Tested by the build-and-run `moulin prod-devel-rcar.yaml --MACHINE h3ulcb-4x2g --GRAPHICS sources --ENABLE_DOMU yes`

lorc commented on meta-xt-prod-devel-rcar-control/recipes-guests/domu/domu.bbappend:
```diff
@@ -27,6 +27,30 @@ FILES:${PN} += " \
+# inside `dtdev = []` construction because we will use
+# linux's echo command to put multiline string into domu.cfg
+python () {
+    if d.getVar("XT_DOMU_CONFIG_NAME", True) == "domu-generic-h3-4x2g.cfg":
```
This looks fragile.

I'd prefer to see use of flags here instead of inline python. Something like

```GSX_DTDEV[h3] = \
    '/soc/gsx_pv0_domu',
    '/soc/gsx_pv1_domu',
    '/soc/gsx_pv2_domu',
    '/soc/gsx_pv3_domu',
```

and then some logic (maybe even inline python, okay) that choose either m3 or h3

lorc commented on meta-xt-prod-devel-rcar-control/recipes-guests/domu/domu.bbappend:
```diff
@@ -45,7 +69,11 @@ do_install:append() {
-        cat ${WORKDIR}/pvr-${XT_DOMU_CONFIG_NAME} >> ${CFG_FILE}
+        if ${@bb.utils.contains('MACHINE_FEATURES', 'gsx', 'true', 'false', d)}; then
+            cat ${WORKDIR}/gsx-common.cfg >> ${CFG_FILE}
+            sed -i "s/cma=32M/cma=256M pvrsrvkm.DriverMode=1/g" ${CFG_FILE}
```
We had many issues with `sed` earlier. I'd prefer some other solution. If you already building part of config dynamically in this recipe, you may build commandline as well.
